### PR TITLE
Check response and query formats in oracle register transaction

### PR DESCRIPTION
**Pull request by gorillainduction**
https://www.pivotaltracker.com/story/show/163287370

### CODE REVIEW

**Comment by gorillainduction** [comment]
Rebased on Minerva again to fix CI

File: apps/aecore/src/aec_tx_processor.erl [hunk 1]
```diff
@@ -347,11 +349,13 @@ eval_one({Op, Args}, S) ->
     end.
 
 new_state(Trees, Height, TxEnv) ->
+    ProtocolVersion = aec_hard_forks:protocol_effective_at_height(Height),
     #state{ trees = Trees
           , cache = dict:new()
           , env   = dict:new()
           , height = Height
           , tx_env = TxEnv
+          , protocol = ProtocolVersion
```
Comment (lucafavatella) [inline apps/aecore/src/aec_tx_processor.erl]: I see derived. Ok.

File: apps/aeoracle/test/aeoracle_SUITE.erl [hunk 2]
```diff
@@ -122,6 +137,21 @@ register_oracle_negative(_Cfg) ->
...
+    RTx8 = aeo_test_utils:register_tx(PubKey, ABISophia#{query_format => <<"foo">>}, S1),
+    RTx9 = aeo_test_utils:register_tx(PubKey, ABISophia#{response_format => <<"foo">>}, S1),
+    RTx10 = aeo_test_utils:register_tx(PubKey, ABISophia, S1),
+    meck:expect(aec_hard_forks, protocol_effective_at_height, fun(_) -> ?ROMA_PROTOCOL_VSN end),
+    ?assertMatch({ok, _}, aetx:process(RTx8, Trees, Env)),
+    ?assertMatch({ok, _}, aetx:process(RTx9, Trees, Env)),
+    meck:expect(aec_hard_forks, protocol_effective_at_height, fun(_) -> ?MINERVA_PROTOCOL_VSN end),
+    ?assertEqual({error, bad_query_format}, aetx:process(RTx8, Trees, Env)),
+    ?assertEqual({error, bad_response_format}, aetx:process(RTx9, Trees, Env)),
+    ?assertMatch({ok, _}, aetx:process(RTx10, Trees, Env)),
```
Comment (lucafavatella) [inline apps/aeoracle/test/aeoracle_SUITE.erl]: All tested - thanks.

File: apps/aecore/src/aec_tx_processor.erl [hunk 3]
```diff
@@ -1031,6 +1035,32 @@ assert_oracle_abi_version(ABIVersion, S) ->
...
+        false ->
+            %% Backwards compatible: no check of this.
+            ok;
+        true ->
+            assert_typerep_format(QFormat, bad_query_format),
+            assert_typerep_format(RFormat, bad_response_format)
+    end.
+
+assert_typerep_format(Format, Error) ->
+    try aeso_heap:from_binary(typerep, Format) of
```
Comment (lucafavatella) [inline apps/aecore/src/aec_tx_processor.erl]: I see same `{ok, TypeRep} = aeso_heap:from_binary(typerep, Format)` as in `assert_oracle_format_match`. Ok.

**Pull request merged**